contributing.md

```diff
@@ -0,0 +1,59 @@
+
+### Contributing by adding an issue
+If you have found a bug and would like to report it or if you have a feature that you feel we should add, then we'd love it if you opened an issue! ❤️ Before you do, please search the other issues to avoid creating a duplicate issue. 
+To submit a new issue just hit the issue button and a choice between two templates should appear. Then, follow along with the template you chose. If you don't know how to fill in all parts of the template go ahead and skip those parts. You can edit the issue later.
+
+### Contributing through code
+In order to contribute through code follow the steps below. Note that you don't need to be the best programmer to contribute. Our discord is open for questions
+
+ 1. **Pick a feature** you would like to add or a bug you would like to fix
+	 - If you wish to contribute but what you want to fix/add is not already covered in an existing issue, please open a new issue
+ 2. **Discuss** the issue with the rest of the community
+	 - Before you write any code, it is recommended that you discuss your intention to write the code on the issue you are attempting to edit
+	 -  This helps to stop you from wasting your time duplicating the work of others that maybe working on the same issue; at the same time
+	 - This step also allows you to get helpful pointers on the community on some problems they may have encountered on similar issues
+ 3. **Fork** the repository
+	 - A fork creates a copy of the code on your Github, so you can work on it separately from everyone else.
+	 - You can learn more about forking [here](https://docs.github.com/en/get-started/quickstart/fork-a-repo)
+ 4. Ensure that you have **commit signing** enabled
+	 - This ensures that the code you submit was committed by you and not someone else who claims to be you
+	 - You can learn more about how to setup commit signing [here](https://www.freecodecamp.org/news/what-is-commit-signing-in-git/ "https://www.freecodecamp.org/news/what-is-commit-signing-in-git/")
+	 - If you have already made some commits that you wish to put in a pull request without signing them, then you can follow [this guide](https://dev.to/jmarhee/signing-existing-commits-with-gpg-5b58) on how to fix that.
+ 5. **Clone** the repository to your your computer.
+	 - This puts a copy of your fork on your computer so you can edit it
+	 - You can learn more about cloning repositories [here](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
+ 6. **Build** the project
+	 - For a detailed look on how to build {{project-name}} look at our [README file](<link to readme>)
+ 7. **Start writing** the code you wanted
+	 - Open up your favorite code editor and make the changes that you wanted to make to the repository
+	 - Make sure to test your code with `{{test-code-cmd}}`
+ 8. **Write tests** for your code
```

Comment:
We should talk about when to have a `unit` vs `integration` test: https://doc.rust-lang.org/book/ch11-03-test-organization.html as well and how that relates to Rust.